docs/docs/testing.md

```diff
@@ -1744,6 +1744,225 @@ Luckily, RedwoodJS has several api testing utilities to make [testing functions
+  await product({ id: scenario.product.three.id })
+
+// Pattern 1: Only check that the data is present in the cache
+  expect(testCacheClient).toHaveCached(scenario.product.three)
+
+// Pattern 2: Check that data is cached, at a specific key
+   expect(testCacheClient).toHaveCached(
+      `cached-product-${scenario.product.three.id}`,
+      scenario.product.three
+    )
+
+// Pattern 3: Check that data is cached, in a key matching the regex
+  expect(testCacheClient).toHaveCached(
+      /cached-.*/,
+      scenario.product.three
+    )
+```
+
+
+:::info Serialized Objects in Cache
+Remember that the cache only ever contains serialized objects. So if you passed an object like this:
+```js
+{
+  id: 5,
+  published: new Date('12/10/1995')
+}
+
+```
+
+The published key will be serialized and stored as a string. To make testing easier for you, we serialize the object you are passing when you use the `toHaveCached` matcher, before we compare it against the value in the cache
+:::
+
+### Partial Matching
+It can be a little tedious to check that every key in the object you are looking for matches. This is especially true if you have autogenerated values such as `updatedAt` and `cuid` Ids.
+
+To help with this, we've provided a helper for partial matching!
+
+```ts
+// highlight-next-line
+import { partialMatch } from '@redwoodjs/testing/cache'
+
+
+scenario('returns all products', async (scenario: StandardScenario) => {
+  await products()
+
+ // Partial match using the toHaveCached, if you supply a key
+   expect(testCacheClient).toHaveCached(
+    /cached-products.*/,
+    // highlight-next-line
+    partialMatch([{ name: 'LS50', brand: 'KEF' }])
+  )
+
+  // Or you can use the .contents getter
+  expect(testCacheClient.contents).toContainEqual(
+  // check that an array contains an object matching
+  // highlight-next-line
+   partialMatch([{ name: 'LS50', brand: 'KEF' }])
+  )
+}
+
+scenario('finds a single product', () = {
+   await product({id: 5})
+
+  // You can also check for a partial match of an object
+   expect(testCacheClient).toHaveCached(
+      /cached-.*/,
+  // highlight-start
+      partialMatch({
+        name: 'LS50',
+        brand: 'KEF'
+      })
+    )
+  // highlight-end
+})
+
+```
+
+Partial match is just syntactic sugar - underneath it just uses Jest's `expect.objectContaining` and `expect.arrayContaining`.
+
+The `partialMatch` helper takes two forms of arguments:
+
+- If you supply an object, you are expecting a partial match of that object
+- If you supply an array of objects, you are expecting an array containing a partial match of each of the objects
+
+
+:::tip
+Note that you cannot use `partialMatch` with toHaveCached without supplying a key!
+
+```ts
+// 🛑 Will never pass!
+expect(testCacheClient).toHaveCached(partialMatch({name: 'LS50'}))
+```
+
+For partial matches, you either have to supply a key to `toHaveCached` or use the `cacheClient.contents` helper.
+:::
+
+
+### Strict Matching
```

Comment:
Did we just get rid of the `{ strict }` option altogether, or is it undocumented for now? My idea with `strict` is that we wouldn't do *any* manipulation on the data, expected or returned. It's up to you to make sure your expected data is formatted to match what's in the cache (JSON.stringified).

So using `contents` seems like an in-between state between ALL the help we give you with `toHaveCached()` and the NO help we would give you with `{ strict }`.

Reply:
Hey, yeah I had to remove it because there's no way to disambiguate the arguments sometimes. For example:

`expect(cacheClient).toHaveCached('string-cached-value', {strict: true})`

vs

`expect(cacheClient).toHaveCached('cache-key-1234', { myData: 'hello'})`

There's no way to determine what's an options object, and what's the actual expected object. 🤷